src/command/pre-release.ts

```diff
@@ -0,0 +1,180 @@
+import * as SemVer from 'semver/classes/semver';
+import Hub from '../hub';
+import settings, { IMultipleRepositoryConfiguration } from '../settings';
+
+const COMMAND = 'pre-release';
+
+const DEFAULT_BRANCH = 'master'; // TODO: get this dynamically?
+const FINALIZE_KEYWORD = 'finalize';
+
+/** set of valid release types */
+const RELEASE_TYPES = [
+    'patch',
+    'minor',
+];
+
+class PreReleaseConfiguration implements IMultipleRepositoryConfiguration {
+    public repos: string[];
+}
+
+interface IPullRequest {
+    head: IPullRequestBranch;
+    number: number;
+    title: string;
+    url: string;
+}
+
+interface IPullRequestBranch {
+    sha: string;
+}
+
+export default function preRelease(vorpal): void {
+    let type;
+
+    vorpal
+        .command(`${COMMAND} <ticket> [type]`)
+        .description(`Merges pull requests across repositories associated with the specified <ticket> and upserts a release of the specified <type> (valid options: ${RELEASE_TYPES.map((t) => `"${t}"`).join(' or ')}, default: "${RELEASE_TYPES[0]}")`)
+        .validate((input) => {
+            type = input.type || RELEASE_TYPES[0];
+            return RELEASE_TYPES.includes(type) ? true : `Invalid release type "${type}"`;
+        })
+        .action(function action(args) {
+            const { ticket } = args;
+            const config = settings.get<PreReleaseConfiguration>(COMMAND);
+            const isFinalizeStep = (ticket === FINALIZE_KEYWORD);
+
+            settings.getAbsoluteRepositoryPaths(config).forEach((cwd) => {
+                let pr;
+
+                if (!isFinalizeStep) {
+                    // 1. Find and merge the pull request.
+                    this.log(`[${cwd}] Checking if pull request for ticket '${ticket}' exists`);
+
+                    const prList = Hub.api<IPullRequest[]>(
+                        [`repos/{owner}/{repo}/pulls?head={owner}:${ticket}&base=${DEFAULT_BRANCH}`],
+                        cwd,
+                    );
+
+                    if (0 === prList.length) {
+                        return;
```

Comment:
Thoughts on throwing an error when there are no pull requests across _any_ of the repos?